src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -600,6 +600,9 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 				});
 			}
 		}).finally(() => {
+			const commonKey = task.getCommonTaskId();
+			liveDependencies.delete(commonKey);
+			encounteredTasks.delete(commonKey);
```

Comment:
This is wrong. 

The whole point of encounteredTasks is that its all tasks that have been started to date, including those that have finished.